src/font.rs

```diff
@@ -118,6 +118,11 @@ impl Font {
     }
 
     fn load_impl(path: &Path, request: DataRequest) -> Result<Font, Error> {
+        if !path.exists() {
+            let msg = format!("invalid path: {:?}", path);
+            return Err(std::io::Error::new(std::io::ErrorKind::NotFound, msg).into());
+        }
+
         let meta_path = path.join(METAINFO_FILE);
         let mut meta: MetaInfo = plist::from_file(meta_path)?;
```

Comment:
You are suggesting to use the first mandatory plist file read at this level and propagate the plist lib read error, possibly modified with path data?

Reply:
D'oh, wait, I think I misread the function. Doing a path check here is fine, but I think it should be done for all file reads. Mh.

Reply:
It looks like there is auto-UFO version up-conversion?  If so, should we add path checks for the cross-version mandatory files:

- metainfo.plist
- glyphs/contents.plist

and not check layercontents.plist in case this is a UFO < v3 source read?  Or check the previously read metainfo data to determine the UFO version and validate mandatory files at this stage?  This PR was initially intended to address typos in source dir paths, but I'd be happy to add some read side spec validations if this seems appropriate here and isn't supported elsewhere in the library.

Reply:
Yes we shoot for the stars because no one has time for UFO < v3. The idea is you read files as required by the version in question and then upgrade at the end. What specifically would you do?

Reply:
Maybe something along these lines?

d9219beeac3eb9de25a63da8963886aa3e6cd1cd

Reply:
https://github.com/linebender/norad/pull/146